x-pack/test/rule_registry/common/lib/authentication/roles.ts

```diff
@@ -0,0 +1,598 @@
+        feature: {
+          siem: ['all'],
+          actions: ['all'],
+          actionsSimulators: ['all'],
+        },
+        spaces: ['*'],
+      },
+    ],
+  },
+};
+
+export const securitySolutionOnlyReadSpacesAll: Role = {
+  name: 'sec_only_read',
+  privileges: {
+    elasticsearch: {
+      indices: [
+        {
+          names: ['*'],
+          privileges: ['all'],
+        },
+      ],
+    },
+    kibana: [
+      {
+        feature: {
+          siem: ['read'],
+          actions: ['read'],
+          actionsSimulators: ['read'],
+        },
+        spaces: ['*'],
+      },
+    ],
+  },
+};
+
+export const observabilityOnlyAllSpacesAll: Role = {
+  name: 'obs_only_all',
+  privileges: {
+    elasticsearch: {
+      indices: [
+        {
+          names: ['*'],
+          privileges: ['all'],
+        },
+      ],
+    },
+    kibana: [
+      {
+        feature: {
+          apm: ['all'],
+          actions: ['all'],
+          actionsSimulators: ['all'],
+        },
+        spaces: ['*'],
+      },
+    ],
+  },
+};
+
+export const observabilityOnlyReadSpacesAll: Role = {
+  name: 'obs_only_read_all_spaces',
+  privileges: {
+    elasticsearch: {
+      indices: [
+        {
+          names: ['*'],
+          privileges: ['all'],
+        },
+      ],
+    },
+    kibana: [
+      {
+        feature: {
+          apm: ['read'],
+          ruleRegistry: ['all'],
+          actions: ['read'],
+          builtInAlerts: ['all'],
+          alerting: ['all'],
+        },
+        spaces: ['*'],
+      },
+    ],
+  },
+};
+
+export const roles = [
+  noKibanaPrivileges,
+  globalRead,
+  securitySolutionOnlyAll,
+  securitySolutionOnlyRead,
+  observabilityOnlyAll,
+  observabilityOnlyRead,
+  observabilityOnlyReadSpacesAll,
+];
+
+/**
+ * These roles are specifically for the security_only tests where the spaces plugin is disabled. Most of the roles (except
+ * for noKibanaPrivileges) have spaces: ['*'] effectively giving it access to the space1 space since no other spaces
+ * will exist when the spaces plugin is disabled.
+ */
+export const rolesDefaultSpace = [
+  noKibanaPrivileges,
+  globalRead,
+  securitySolutionOnlyAllSpacesAll,
+  securitySolutionOnlyReadSpacesAll,
+  observabilityOnlyAllSpacesAll,
+  observabilityOnlyReadSpacesAll,
+];
+
+/**
+ * These roles are only to be used in the 'trial' tests
+ * since they rely on subfeature privileges which are a gold licencse feature
+ * maybe put these roles into a separate roles file like "trial_roles"?
+ */
+export const observabilityMinReadAlertsRead: Role = {
+  name: 'obs_only_alerts_read',
+  privileges: {
+    elasticsearch: {
+      indices: [
+        {
+          names: ['*'],
+          privileges: ['all'],
+        },
+      ],
+    },
+    kibana: [
+      {
+        feature: {
+          apm: ['minimal_read', 'alerts_read'],
+          ruleRegistry: ['all'],
+          actions: ['read'],
+          builtInAlerts: ['all'],
+          alerting: ['all'],
```

Comment:
These roles are being used to test authZ that is solely based on the `apm` feature privilege. You should remove unnecessary privileges from these roles, e.g.,

```suggestion
          apm: ['minimal_read', 'alerts_read'],
```

Same goes for all the rest of the roles.